framework/include/controls/ControlInterface.h

```diff
@@ -129,98 +86,59 @@ class ControlInterface
   const std::string & _ci_name;
 
   // Only these objects are allowed to construct this interface
-  friend class RealParameterReporter;
+  friend class RealControlParameterReporter;
   friend class Control;
 };
 
 template<typename T>
-T &
+ControllableParameter<T>
 ControlInterface::getControlParam(const std::string & name)
 {
-  return getControlParamByName<T>(_ci_parameters.get<std::string>(name));
+  return getControlParamByName<T>(_ci_parameters.get<std::string>(name),
+                                  std::numeric_limits<unsigned int>::max());
 }
 
 template<typename T>
-T &
-ControlInterface::getControlParamByName(const std::string & name)
+ControllableParameter<T>
+ControlInterface::getControlParam(const std::string & name, unsigned int count)
```

Comment:
Count allows you to throw an error if you get an unexpected number of controls. If you are only expecting to control one parameter, but you provide a loose name you may be controlling 10, but would never know it.